compute_endpoint/tests/integration/endpoint/executors/test_mpiengine.py

```diff
@@ -0,0 +1,131 @@
+import random
+import uuid
+from queue import Queue
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_endpoint.engines import GlobusMPIEngine
+from globus_compute_sdk.serialize import ComputeSerializer
+from parsl.launchers import SimpleLauncher
+from parsl.providers import LocalProvider
+from tests.utils import ez_pack_function, get_env_vars
+
+temporary_skip = False
+try:
+    from globus_compute_sdk.sdk.bash_function import BashResult
+    from globus_compute_sdk.sdk.mpi_function import MPIFunction
+except ImportError:
+    temporary_skip = True
+
+
+@pytest.fixture
+def nodeslist(num=100):
+    limit = random.randint(1, num)
+    yield [f"NODE{node_i}" for node_i in range(1, limit)]
+
+
+@pytest.fixture
+def mpi_engine(tmp_path, nodeslist):
```

Comment:
Consider moving this under the `engine_runner` fixture to be consistent with other engines.

Reply:
Hey @rjmello, I've moved the fixture. Could you take a look please?

Reply:
@chris-janidlo and I just talked about this. The consensus is that moving this under engine_runner should be done in a follow up PR. The `mpi_engine` fixture needs to return the engine and the nodelist, which doesn't match the engine_runner, and will likely touch all the tests that use the runner. It would be cleaner to split that into it's own PR.